core/src/main/java/com/linecorp/armeria/internal/client/AbstractRuleBuilderUtil.java

```diff
@@ -135,7 +142,8 @@ private boolean applySlow(ClientRequestContext ctx, @Nullable Throwable cause, R
                 }
             }
 
-            return false;
+            return responseDurationFilter != null &&
```

Comment:
- Don't we need to change `RequestLogProperty.RESPONSE_TRAILERS` to `RequestLogProperty.RESPONSE_END_TIME`?
https://github.com/line/armeria/blob/2bb48ade4cbdcfd10506003fb7420b20cfd97f55/core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java#L377-L378
https://github.com/line/armeria/blob/2bb48ade4cbdcfd10506003fb7420b20cfd97f55/core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerClient.java#L335-L336
- Should we add `log.isAvailable(RequestLogProperty.RESPONSE_END_TIME)` to the condition just in case?

Reply:
> Don't we need to change RequestLogProperty.RESPONSE_TRAILERS to RequestLogProperty.RESPONSE_END_TIME?

In the code below, I checked that `responseTrailers(log.responseTrailers())` set RequestLogProperty.RESPONSE_TRAILERS value first and `endResponse0(...)` set RequestLogProperty.RESPONSE_END_TIME value. Then, is the below correct?
```java
log().whenAvailable(RESPONSE_END_TIME).thenAccept(log ->  {
    // Here, can I ensure that `log` has response trailers?
    ...
})
```

https://github.com/line/armeria/blob/ce5b42f36a4a5515a28cee7a15b2a3e2ca5f76c2/core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java#L642-L648

> Should we add log.isAvailable(RequestLogProperty.RESPONSE_END_TIME) to the condition just in case?

Let me add that.

Reply:
Updated!